src/hotspot/share/opto/loopopts.cpp

```diff
@@ -2867,8 +2867,7 @@ ProjNode* PhaseIdealLoop::insert_if_before_proj(Node* left, bool Signed, BoolTes
 
   int opcode = iff->Opcode();
   assert(opcode == Op_If || opcode == Op_RangeCheck, "unexpected opcode");
-  IfNode* new_if = (opcode == Op_If) ? new IfNode(proj2, bol, iff->_prob, iff->_fcnt):
-    new RangeCheckNode(proj2, bol, iff->_prob, iff->_fcnt);
```

Comment:
Since I re-used this pattern again, I extracted this into `make_with_same_profile()`.

Reply:
Do you want to take over the assert as well?

Reply:
I think it should now be covered with the additional assert in `make_with_same_profile()`. If we pass something else, we will hit that assert.